aikau/src/main/resources/alfresco/forms/ControlRow.js

```diff
@@ -179,11 +179,38 @@ define(["alfresco/layout/HorizontalWidgets",
        * @param {object} widget The widget to validate
        * @param {number} index The index of the widget to validate
        */
-      validateChildFormControlValue: function alfresco_forms_ControlRow__validateChildFormControlValue(widget, index) {
+      validateChildFormControlValue: function alfresco_forms_ControlRow__validateChildFormControlValue(widget, /*jshint unused:false*/ index) {
          if (typeof widget.validateFormControlValue === "function")
          {
             widget.validateFormControlValue();
          }
+      },
+
+      /**
+       * Iterates over the child form controls and publishes the value of each one.
+       * 
+       * @instance
+       * @param {Deferred} deferred A deferred object can optionally be passed. This will only be resolved as widget value
+       */
+      publishValue: function alfresco_forms_ControlRow__publishValue(deferred) {
+         array.forEach(this._processedWidgets, lang.hitch(this, this.publishChildValue, deferred));
+      },
+
+      /**
+       * This is called by the [publishValue]{@link module:alfresco/forms/ControlRow#publishValue} function for each
+       * of the child form controls in the row and calls its 
+       * [publishValue]{@link module:alfresco/forms/controls/BaseFormControl#publishValue} function.
+       * 
+       * @instance
+       * @param {Deferred} deferred A deferred object can optionally be passed. This will only be resolved as widget value
```

Comment:
See above re: square brackets. Is an object of type "{Deferred}" defined anywhere?